packages/flutter_tools/lib/src/ios/devices.dart

```diff
@@ -810,12 +828,42 @@ class IOSDeviceLogReader extends DeviceLogReader {
   // Sometimes (race condition?) we try to send a log after the controller has
   // been closed. See https://github.com/flutter/flutter/issues/99021 for more
   // context.
-  void _addToLinesController(String message) {
+  void _addToLinesController(String message, IOSDeviceLogSource source) {
     if (!linesController.isClosed) {
+      if (_excludeLog(message, source)) {
+        return;
+      }
       linesController.add(message);
     }
   }
 
+  /// Used to track messages prefixed with "flutter:" when [useBothLogDeviceReaders]
+  /// is true.
+  final List<String> _streamFlutterMessages = <String>[];
+
+  /// When using both `idevicesyslog` and `ios-deploy`, exclude logs with the
+  /// "flutter:" prefix if they have already been added to the stream. This is
+  /// to prevent duplicates from being printed.
+  ///
+  /// If a message does not have the prefix, exclude it if the message's
+  /// source is `idevicesyslog`. This is done because `ios-deploy` and
+  /// `idevicesyslog` often have different prefixes on non-flutter messages
+  /// and are often not critical for CI tests.
+  bool _excludeLog(String message, IOSDeviceLogSource source) {
```

Comment:
FYI: I added this change because when `ios-deploy` fails to return the Dart VM url, it usually doesn't return any logs from the app. So I added this check so we can get other important logs from `idevicesyslog` other than just the Dart VM url. This way tests that use the device logs to check things (such as `microbenchmarks_ios`) will also work with this workaround.